mediar-ai/screenpipe
# fix: conditionally add Authorization header for screenpipe-cloud AI p…

## Description
Fix authorization header handling in search-chat component. Previously, we were always sending the Authorization header with the user token regardless of provider type. This change makes it conditional - only sending the Authorization header when using the screenpipe-cloud provider.

The reason for this change is that when using direct OpenAI provider, we don't want to send our internal user token as it conflicts with OpenAI's authentication. The screenpipe-cloud provider needs our user token for authentication, while OpenAI direct needs its own API key through different means.

related issue: N/A

## How to test

1. Configure screenpipe with OpenAI direct provider
   - Verify requests work without Authorization header
   - Verify OpenAI API calls succeed
2. [NEEDS TESTING] Configure screenpipe with screenpipe-cloud provider
   - Verify requests include Authorization header with user token
   - Verify API calls succeed through screenpipe-cloud

⚠️ Note: This PR needs testing with screenpipe-cloud provider before merging.
